src/MetadataUtility/Models/Sensor.cs

```diff
@@ -20,12 +18,12 @@ public record Sensor
         /// <summary>
         /// Gets the firmware version of this sensor.
         /// </summary>
-        public string Firmware { get; init; }
+        public float Firmware { get; init; }
```

Comment:
```suggestion
        public string Firmware { get; init; }
```

You cannot guarantee a  firmware version is parseable as a float. FL do adhere to the 'version is a number' convention, but most versions is the world can have multiple dotted segments.